src/lib.rs

```diff
@@ -851,103 +854,272 @@ pub struct RegisteredClaims {
+            Validation::Validate(temporal_options) => {
+                let now = temporal_options.now.unwrap_or(Utc::now());
+
+                match self.expiry {
+                    Some(Timestamp(expiry)) if now - expiry > temporal_options.epsilon => {
+                        Err(ValidationError::Expired(now - expiry))
+                    }
+                    _ => Ok(()),
+                }
+            }
+        }
     }
 
-    /// Check `a` is after `b` within a tolerated duration of `e`, where `e` is unsigned: a - b >= -e
-    fn is_after<Tz, Tz2>(a: DateTime<Tz>, b: DateTime<Tz2>, e: std::time::Duration) -> Result<bool, ValidationError>
-    where
-        Tz: chrono::offset::TimeZone,
-        Tz2: chrono::offset::TimeZone,
-    {
-        // FIXME: `chrono::Duration` is a re-export of `time::Duration` and this returns has an error of type
-        // `time::OutOfRangeError`. We don't want to put `time` as a dependent crate just to `impl From` for this...
-        // So I am just going to `map_err`.
-        use std::error::Error;
+    ///Validates that if the token has an `nbf` claim, it has passed.
+    pub fn validate_nbf(&self, validation: Validation<TemporalOptions>) -> Result<(), ValidationError> {
+        match validation {
+            Validation::Ignored => Ok(()),
+            Validation::Validate(temporal_options) => {
+                let now = temporal_options.now.unwrap_or(Utc::now());
 
-        let e = chrono::Duration::from_std(e).map_err(|e| ValidationError::TemporalError(e.description().to_string()))?;
-        Ok(a.signed_duration_since(b) >= -e)
+                match self.not_before {
+                    Some(Timestamp(nbf)) if nbf - now > temporal_options.epsilon => {
+                        Err(ValidationError::NotYetValid(nbf - now))
+                    }
+                    _ => Ok(()),
+                }
+            }
+        }
     }
 
-    /// Check that `a` is before `b` within a tolerated duration of `e`, where `e` is unsigned: a - b <= e
-    fn is_before<Tz, Tz2>(a: DateTime<Tz>, b: DateTime<Tz2>, e: std::time::Duration) -> Result<bool, ValidationError>
-    where
-        Tz: chrono::offset::TimeZone,
-        Tz2: chrono::offset::TimeZone,
-    {
-        // FIXME: `chrono::Duration` is a re-export of `time::Duration` and this returns has an error of type
-        // `time::OutOfRangeError`. We don't want to put `time` as a dependent crate just to `impl From` for this...
-        // So I am just going to `map_err`.
-        use std::error::Error;
+    ///Validates that if the token has an `iat` claim, it is not in the future and not older than the Duration
+    pub fn validate_iat(&self, validation: Validation<(Duration, TemporalOptions)>) -> Result<(), ValidationError> {
+        match validation {
+            Validation::Ignored => Ok(()),
+            Validation::Validate((max_age, temporal_options)) => {
+                let now = temporal_options.now.unwrap_or(Utc::now());
```

Comment:
Use [`unwrap_or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_else) for lazy evaluation.